### DISCUSSION TITLE
Keycloak and the ability to validate client x5c

### DISCUSSION THREAD

**Discussion by genie137** [keycloak/keycloak#34935] [Q&A]
We are currently looking at Keycloak as an Authorization Server for an OAuth2.0 implementation within our organization.
However we have an requirement for the implementation that we cannot verify using the online documentation.
We are planning to use the client credential flow with private-key-jwt authorization, in which the public key is shared by the client using JWK.
 It is a requirement that the public key is dynamically generated, and thus cannot be registered in Keycloak before requesting an access-token.
However we also have the requirement that we need to validate the root-ca that the public key was generated with.
 Unfortunately this is not included in the OAuth2.0 spec.
 On the client side we can include the x5c claim in the JWK as described in the spec here: https://datatracker.ietf.org/doc/html/rfc7517#section-4.7.
Does Keycloak have the ability to validate that a certificate in the x5c chain is equal to a given certificate, or a certificate inside the truststore?

**Comment by Lingkar** [comment]
@genie137 Would you mind giving a follow-up on your findings? Did you resolve your requirement with/without KeyCloak?

Thanks for your time.
